ocw/dataset.py

```diff
@@ -360,20 +363,63 @@ def end(self, value):
 
         self._end = value
 
-    def __str__(self):
-        lat_range = "({}, {})".format(self._lat_min, self._lat_max)
-        lon_range = "({}, {})".format(self._lon_min, self._lon_max)
-        temporal_boundaries = "({}, {})".format(self._start, self._end)
+def shapefile_boundary(boundary_type, region_names):
+    '''
+    :param boundary_type: The type of spatial subset boundary
+    :type boundary_type: :mod:'string'
 
-        formatted_repr = (
-            "<Bounds - "
-            "lat-range: {}, "
-            "lon-range: {}, "
-            "temporal_boundaries: {}> "
-        )
+    :param region_names: An array of regions for spatial subset
+    :type region_names: :mod:'list'
+    '''
 
-        return formatted_repr.format(
-            lat_range,
-            lon_range,
-            temporal_boundaries,
-        )
+    map_read = Basemap()
+    regions =[]
+    if boundary_type == 'us_states':
+        map_read.readshapefile(ocw.__path__[0]+'/shape/usa_states','usa_states')
+        #map_read.readshapefile('/home/huikyole/climate/ocw'+'/shape/usa_states','usa_states')
+        for region_name in region_names:
+            for iregion, region_info in enumerate(map_read.usa_states_info):
+                if region_info['st'] == region_name:
+                    regions.append(numpy.array(map_read.usa_states[iregion]))
+    if boundary_type == 'countries':
+        map_read.readshapefile(ocw.__path__[0]+'/shape/countries','countries')
+        #map_read.readshapefile('/home/huikyole/climate/ocw'+'/shape/countries','countries')
+        for region_name in region_names:
+            for iregion, region_info in enumerate(map_read.countries_info):
+                if region_info['COUNTRY'] == region_name:
+                    regions.append(numpy.array(map_read.countries[iregion]))
+    return regions
+
```

Comment:
Please add the shapefiles directory to this PR. Without it the unit tests will fail. I will also eventually need to test if this works with conda / pypi packaging since this may require changes to the setup script.